### PR TITLE
Better handle vendored trees

### PR DESCRIPTION
**Pull request by jwendell**
- Skip the vendor dir in `find` usages
- Create a target for running `go mod download` to be reused by other
  projects, that work on vendored trees.

### CODE REVIEW

**Comment by jwendell** [comment]
Part of https://github.com/istio/istio/issues/25330

File: files/common/Makefile.common.mk [hunk 1]
```diff
@@ -64,15 +64,17 @@ lint-typescript:
...
+lint-licenses: mod-download-go
 	@license-lint --config common/config/license-lint.yml
 
 lint-all: lint-dockerfiles lint-scripts lint-yaml lint-helm lint-copyright-banner lint-go lint-python lint-markdown lint-sass lint-typescript lint-protos lint-licenses
 
 tidy-go:
 	@go mod tidy
 
+mod-download-go:
+	@-GOFLAGS="-mod=readonly" go mod download
```
Comment (howardjohn) [inline files/common/Makefile.common.mk:76]: For my understanding, can you explain why we need mod=readonly? My understanding is that makes the go.mod file not modified here. Maybe that is useful, but I don't see how it relates to vendor directory. Does it somehow?

Comment (jwendell) [reply files/common/Makefile.common.mk:76]: in go 1.14, if there's a vendor dir, `-mod=vendor` is implied unless explicitly overwritten.
also, users may have the `GOFLAGS` globally set with `-mod=vendor`.

So, overwriting this with `readonly` on the command line seems to be the safest thing to do, that works on both scenarios without any side effect (like updating go.mod).

Comment (jwendell) [reply files/common/Makefile.common.mk:76]: also, tools like `go list -m all` do not work in vendor mode. After this gets merged/propagated I'll submit a PR for istio fixing it.

Comment (howardjohn) [reply files/common/Makefile.common.mk:76]: Oh so this more about unsettting GOFLAGs than it is about setting it to readonly?

Comment (jwendell) [reply files/common/Makefile.common.mk:76]: Yes, the choices are `readonly` or `mod`. I find readonly more conservative, less intrusive.

**Pull request merged**